tpp-app/src/services/LogSymptomsService.js

```diff
@@ -84,41 +89,52 @@ export const LogMultipleDayPeriod = async (dates) => new Promise(async (resolve,
 
             })
 
-            if(calendarData[curYear]){
-                AsyncStorage.setItem(curYear.toString(), JSON.stringify(calendarData[curYear]))
-                .then(() => resolve())
-                .catch((e) => {
-                    console.log(JSON.stringify(e));
-                    reject(`Unable to mergeItem and post symptoms for multiselect.`);
-                });
-                
-            }
-            
-            if(calendarData[curYear - 1]){
-                AsyncStorage.setItem((curYear - 1).toString(), JSON.stringify(calendarData[curYear - 1]))
-                .then(() => resolve())
-                .catch((e) => {
-                    reject(`Unable to mergeItem and post symptoms for multiselect.`);
-                    console.log(JSON.stringify(e));
-                });
+            for (const [key, value] of Object.entries(calendarData)){
+                // console.log(key, value);
+                if(value){
+                    try {
+                        await AsyncStorage.setItem(key.toString(), JSON.stringify(value));
+                    } catch (error) {
+                        console.log(`LogMultipleDayPeriod error: ${JSON.stringify(error)}`)
+                    }
+                }
             }
 
-            // a bit unneccessary since you can't log symptoms for the future.
-            if(calendarData[curYear + 1]){
-                AsyncStorage.setItem((curYear + 1).toString(), JSON.stringify(calendarData[curYear + 1]))
-                .then(() => resolve())
-                .catch((e) => {
-                    reject(`Unable to mergeItem and post symptoms for multiselect.`);
-                    console.log(JSON.stringify(e));
-                });
-            }
+            // if(calendarData[curYear]){
```

Comment:
Why not just delete this?

Reply:
True I totally forgot about it when I fixed it lol

Reply:
removed!